frontend/src/pages/library/_viewId.vue

```diff
@@ -57,258 +65,224 @@ import { getMusicGenresApi } from '@jellyfin/sdk/lib/utils/api/music-genres-api'
-    },
-    onChangeFilter(filter: Record<string, [string]>): boolean | void {
-      this.hasFilters = Object.values(filter).some((value) => {
-        return value.length > 0;
-      });
+const hasViewTypes = computed(
+  () =>
+    library.value &&
+    library.value.CollectionType !== undefined &&
+    library.value.CollectionType !== null &&
+    library.value.CollectionType !== 'homevideos'
+);
 
-      this.genresFilter = filter.genres;
-      this.statusFilter = filter.status;
-      this.yearsFilter = filter.years;
-      this.ratingsFilter = filter.ratings;
-      this.filterHasSubtitles = filter.features.includes('HasSubtitles');
-      this.filterHasTrailer = filter.features.includes('HasTrailer');
-      this.filterHasSpecialFeature =
-        filter.features.includes('HasSpecialFeature');
-      this.filterHasThemeSong = filter.features.includes('HasThemeSong');
-      this.filterHasThemeVideo = filter.features.includes('HasThemeVideo');
+const isSortable = computed(
+  () =>
+    viewType.value &&
+    // Not everything is sortable, so depending on what we're showing, we need to hide the sort menu.
+    // Reusing this as "isFilterable" too, since these seem to go hand in hand for now.
+    ![
+      'MusicArtist',
+      'Person',
+      'Genre',
+      'MusicGenre',
+      'MusicGenre',
+      'Studio'
+    ].includes(viewType.value)
+);
 
-      // The following technically have a "false" state for excluding them.
-      // TODO: Maybe add exclusion to the filters
-      this.filterIsHd = filter.types.includes('isHD') ? true : undefined;
+const sortBy = ref<string>();
+const sortAscending = ref(true);
+function onChangeSort(sort: string, ascending: boolean): void {
+  sortBy.value = sort;
+  sortAscending.value = ascending;
+}
 
-      this.filterIs4k = filter.types.includes('is4K') ? true : undefined;
+const filters = ref<Filters>({
+  status: [],
+  features: [],
+  genres: [],
+  ratings: [],
+  types: [],
+  years: []
+});
+function onChangeFilter(changedFilters: Filters): void {
+  filters.value = changedFilters;
+}
+const hasFilters = computed(() =>
+  Object.values(filters.value).some(({ length }) => length > 0)
+);
 
-      this.filterIs3d = filter.types.includes('is3D') ? true : undefined;
+watch([library, sortBy, sortAscending, viewType, filters], refreshItems);
 
-      this.refreshItems();
-    },
-    async refreshItems(): Promise<void> {
-      this.loading = true;
+async function fetchLibrary(itemId: string) {
+  loadingLibrary.value = true;
+  try {
+    library.value =
+      (
+        await remote.sdk.newUserApi(getItemsApi).getItems({
+          userId: remote.auth.currentUserId,
+          ids: [itemId]
+        })
+      ).data?.Items?.[0] ?? undefined;
+  } catch (error) {
+    console.error(error);
+    useSnackbar(t('failedToRefreshItems'), 'error');
+  } finally {
+    loadingLibrary.value = false;
+  }
+}
 
-      try {
-        let itemsResponse;
+async function refreshItems(): Promise<void> {
```

Comment:
Said the ``ReferenceError`` thing above and then I discover this function is defined after the watcher :)
See what I mean?